crates/lang/src/reflect/trait_def/registry.rs

```diff
@@ -61,3 +67,37 @@ where
 {
     type Env = E;
 }
+
+impl<E> ToAccountId<E> for TraitDefinitionRegistry<E>
+where
+    E: ink_env::Environment,
+{
+    /// `to_account_id` is not allowed for `TraitDefinitionRegistry`.
+    ///
+    /// We insert markers for these errors in the generated contract code.
+    /// This is necessary since we can't check these errors at compile time
+    /// of the contract.
+    /// `cargo-contract` checks the contract code for these error markers
+    /// when building a contract and fails if it finds markers.
```

Comment:
```suggestion
    /// The trait methods which are implemented on the `TraitDefinitionRegistry` should not be
    /// called directly. Instead, users should opt to go through the `__ink_TraitInfo` associated
    /// type instead.
    ///
    /// In order to enforce this we insert these error markings in the generated contract code. The
    /// `cargo-contract` tool will check for these markers during its build step and fail if they
    /// are found.
    ///
    /// We do this since we can't check for these errors at compile time. 
```